pkg/prometheus/promcfg.go

```diff
@@ -907,7 +914,11 @@ func (cg *ConfigGenerator) generateProbeConfig(
 	}
 
 	assetKey := fmt.Sprintf("probe/%s/%s", m.Namespace, m.Name)
-	cfg = addOAuth2ToYaml(cfg, version, m.Spec.OAuth2, oauth2Secrets, assetKey)
+	cfg = addOAuth2ToYaml(cfg, version, m.Spec.OAuth2, store.OAuth2Assets, assetKey)
+
+	if version.GTE(semver.MustParse("2.26.0")) {
```

Comment:
I would do this check inside `addAuthorizationToYaml` to make sure it is applied everywhere, as well as to reduce code duplication.